backend_new/test/unit/services/application-flagged-set.service.spec.ts

```diff
@@ -0,0 +1,2431 @@
+          },
+        ],
+      },
+    });
+
+    expect(prisma.applicationFlaggedSet.count).toHaveBeenNthCalledWith(2, {
+      where: {
+        listingId: 'example id',
+        status: FlaggedSetStatusEnum.pending,
+      },
+    });
+  });
+
+  it('should get a flagged set', async () => {
+    prisma.applicationFlaggedSet.findUnique = jest.fn().mockResolvedValue({
+      id: 'example id',
+    });
+    expect(await service.findOne('example id')).toEqual({
+      id: 'example id',
+    });
+
+    expect(prisma.applicationFlaggedSet.findUnique).toHaveBeenCalledWith({
+      include: {
+        applications: {
+          include: {
+            applicant: true,
+          },
+        },
+        listings: true,
+      },
+      where: {
+        id: 'example id',
+      },
+    });
+  });
+
+  it('should error getting a flagged set that does not exist', async () => {
+    prisma.applicationFlaggedSet.findUnique = jest.fn().mockResolvedValue(null);
+    await expect(
+      async () => await service.findOne('example id'),
+    ).rejects.toThrowError(
+      'applicationFlaggedSetId example id was requested but not found',
+    );
+
+    expect(prisma.applicationFlaggedSet.findUnique).toHaveBeenCalledWith({
+      include: {
+        applications: {
+          include: {
+            applicant: true,
+          },
+        },
+        listings: true,
+      },
+      where: {
+        id: 'example id',
+      },
+    });
+  });
+
+  it('should update showConfirmationAlert', async () => {
+    prisma.applicationFlaggedSet.findFirst = jest.fn().mockResolvedValue({
+      id: 'example id',
+    });
+    prisma.applicationFlaggedSet.update = jest.fn().mockResolvedValue({
+      id: 'example id',
+    });
+    expect(await service.resetConfirmationAlert('example id')).toEqual({
+      success: true,
+    });
+
+    expect(prisma.applicationFlaggedSet.findFirst).toHaveBeenCalledWith({
+      where: {
+        id: 'example id',
+      },
+    });
+
+    expect(prisma.applicationFlaggedSet.update).toHaveBeenCalledWith({
+      data: {
+        showConfirmationAlert: false,
+      },
+      where: {
+        id: 'example id',
+      },
+    });
+  });
+
+  it('should error updating showConfirmationAlert for flagged set that does not exist', async () => {
+    prisma.applicationFlaggedSet.findFirst = jest.fn().mockResolvedValue(null);
+    await expect(
+      async () => await service.resetConfirmationAlert('example id'),
+    ).rejects.toThrowError(
+      'applicationFlaggedSet example id was requested but not found',
+    );
+
+    expect(prisma.applicationFlaggedSet.findFirst).toHaveBeenCalledWith({
+      where: {
+        id: 'example id',
+      },
+    });
+  });
+
+  it('should grab meta data', async () => {
+    const mockCount = jest.fn();
+    mockCount.mockResolvedValueOnce(1);
+    mockCount.mockResolvedValueOnce(2);
+    mockCount.mockResolvedValueOnce(3);
+    mockCount.mockResolvedValueOnce(4);
+    mockCount.mockResolvedValueOnce(5);
```

Comment:
I don't know if it is any cleaner, but jest.fn() is a chainable call so this could be 
```
    const mockCount = jest
      .fn()
      .mockResolvedValueOnce(1)
      .mockResolvedValueOnce(2)
      .mockResolvedValueOnce(3)
      .mockResolvedValueOnce(4)
      .mockResolvedValueOnce(5);

Reply:
updated!